src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +31,257 @@ public void GetSerializableMembers_Interface()
+            yield return new object[] { typeof(int).MakePointerType() };
+            yield return new object[] { typeof(int).MakeByRefType() };
+            yield return new object[] { typeof(GenericClass<>).GetTypeInfo().GenericTypeParameters[0] };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_NotSupportedType_TestData))]
+        public void GetUninitializedObject_NotSupportedType_ThrowsArgumentException(Type type)
+        {
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [InlineData(typeof(AbstractClass))]
+        [InlineData(typeof(StaticClass))]
+        [InlineData(typeof(Interface))]
+        [InlineData(typeof(Array))]
+        public void GetUninitializedObject_AbstractClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        private abstract class AbstractClass { }
+        private static class StaticClass { }
+        private interface Interface { }
+
+        public static IEnumerable<object[]> GetUninitializedObject_OpenGenericClass_TestData()
+        {
+            yield return new object[] { typeof(GenericClass<>) };
+            yield return new object[] { typeof(GenericClass<>).MakeGenericType(typeof(GenericClass<>)) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
```

Comment:
I am still not very happy about testing these internal CoreLib types. The tests that do it are high maintenance. They have to be fixed in creative ways again and again. I do not think it is the right tradeoff.

If you would like more interesting cases covered here, you can add this instead of ``ByRefeference<T>`:

```
struct Reader
{
     Span<byte> _bytes;
     int position;
}
```

This is a public normal type, no runtime internals.